### PR TITLE
support selecting deployment target for app commands

### PR DESCRIPTION
**Pull request by d-g-town**

### CODE REVIEW

File: cli/cmd/commands/app.go [hunk 1]
```diff
@@ -58,6 +59,14 @@ func registerCommand_App(cliConf config.CLIConfig) *cobra.Command {
 		Short: "Runs a command for your application.",
 	}
 
+	appCmd.PersistentFlags().StringVarP(
+		&deploymentTarget,
+		"target",
+		"x",
+		"default",
+		"the deployment target for the app, default is \"default\"",
```
Comment (ianedwards) [inline cli/cmd/commands/app.go:67]: not sure this matters at the moment, but aren't names scoped to project so this default name wouldn't necessarily be true for multi cluster?

File: cli/cmd/v2/update_image.go [hunk 2]
```diff
@@ -9,21 +9,16 @@ import (
...
-	targetResp, err := client.DefaultDeploymentTarget(ctx, projectId, clusterId)
-	if err != nil {
-		return "", fmt.Errorf("error calling default deployment target endpoint: %w", err)
-	}
-
-	if targetResp.DeploymentTargetID == "" {
-		return "", errors.New("deployment target id is empty")
+func UpdateImage(ctx context.Context, tag string, client api.Client, projectId, clusterId uint, appName string, deploymentTargetName string) (string, error) {
+	if deploymentTargetName == "" {
+		return "", errors.New("please provide a deployment target")
```
Comment (ianedwards) [inline cli/cmd/v2/update_image.go:14]: is the default passed in by the caller?

**Review by ianedwards — APPROVED** [review]
any reason we can't do this for apply?

File: cli/cmd/v2/update_image.go (hunk 2 above)
Comment (d-g-town) [reply cli/cmd/v2/update_image.go:14]: yes, it is

File: cli/cmd/commands/app.go (hunk 1 above)
Comment (d-g-town) [reply cli/cmd/commands/app.go:67]: that's a great point, I'll have to think about what the most intuitive way to do this for multicluster is. I'll make adjustments in another PR.

**Comment by d-g-town** [comment]
No reason we can't do it for apply, but I think that might be slightly more involved, so I'll tackle it in another PR

**Pull request merged**